src/commissaire_http/handlers/clusters.py

```diff
@@ -25,11 +29,222 @@ def list_clusters(message, bus):
+            'Cluster', cluster.to_dict()])
+    return create_response(message['id'], response['result'])
+
+
+def check_cluster_member(message, bus):
+    """
+    Checks is a member is part of the cluster.
+
+    :param message: jsonrpc message structure.
+    :type message: dict
+    :param bus: Bus instance.
+    :type bus: commissaire_http.bus.Bus
+    :returns: A jsonrpc structure.
+    :rtype: dict
+    """
+    try:
+        cluster = bus.request('storage.get', params=[
+            'Cluster', {'name': message['params']['name']}, True])
+        if message['params']['host'] in cluster['result']['hostset']:
+            # Return back the host in a list
+            return create_response(message['id'], [message['params']['host']])
+        else:
+            return create_response(
+                message['id'],
+                error='The requested host is not part of the cluster.',
+                error_code=JSONRPC_ERRORS['NOT_FOUND'])
+    except Exception as error:
+        return create_response(
+            message['id'],
+            error=error,
+            error_code=JSONRPC_ERRORS['INTERNAL_ERROR'])
+
+
+def add_cluster_member(message, bus):
+    """
+    Adds a member to the cluster.
+
+    :param message: jsonrpc message structure.
+    :type message: dict
+    :param bus: Bus instance.
+    :type bus: commissaire_http.bus.Bus
+    :returns: A jsonrpc structure.
+    :rtype: dict
+    """
+    try:
+        cluster = bus.request('storage.get', params=[
+            'Cluster', {'name': message['params']['name']}, True])
+        if message['params']['host'] in cluster['result']['hostset']:
+            # Return back the host in a list ... it's already there
+            return create_response(message['id'], [message['params']['host']])
+        else:
+            cluster['result']['hostset'].append(message['params']['host'])
+            bus.request('storage.save', params=[
+                'Cluster', cluster['result'], True])
+
+            return create_response(
+                message['id'],
+                [message['params']['host']]
+            )
+    except Exception as error:
+        return create_response(
+            message['id'],
+            error=error,
+            error_code=JSONRPC_ERRORS['NOT_FOUND'])
+
+
+def delete_cluster_member(message, bus):
+    """
+    Deletes a member from the cluster.
+
+    :param message: jsonrpc message structure.
+    :type message: dict
+    :param bus: Bus instance.
+    :type bus: commissaire_http.bus.Bus
     :returns: A jsonrpc structure.
     :rtype: dict
     """
-    clusters = bus.request('storage.get', 'get', params=['Clusters', {}])
-    response_msg = []
-    for cluster in clusters['clusters']:
-        response_msg.append(cluster['name'])
-    return create_response(message['id'], response_msg)
+    try:
+        cluster = bus.request('storage.get', params=[
+            'Cluster', {'name': message['params']['name']}, True])
```

Comment:
The `FIXME` comment in the old `util.etcd_cluster_remove_host` code is still relevant, I think. Would you mind pasting it here so it doesn't get forgotten?